# <font color='#4682B4'>Capstone Project</font>
## <font color='#4682B4'>“Digital Wellness: Exploring Screen Time, Sleep, and Stress through Data Analysis”</font>
<hr style="height:2px;border:none;color:#4682B4;background-color:#4682B4;" />

### <font color='#4682B4'>Project Objective</font>
This project aims to explore how screen time, sleep habits, and stress levels are connected by carrying out a full exploratory data analysis (EDA) in Python. The work will involve:

* Loading the dataset and getting an initial sense of its structure.
* Cleaning and preparing the data by fixing missing values, correcting formats, and creating useful new columns.
* Running descriptive and exploratory analysis to spot patterns, correlations, and trends.
* Creating clear visualizations with Python libraries like Matplotlib, Seaborn, and Plotly to make the findings easy to understand.
* Summarizing the key insights with straightforward explanations, highlighting important relationships and observations.

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns

### <font color='#4682B4'>1. Dataset Loading</font>
<hr style="height:1.75px;border:none;color:#4682B4;background-color:#4682B4;" />

In [3]:
data = pd.read_csv("Smartphone Usage  Dataset.csv")

### <font color='#4682B4'>2. Initial Overview</font>
<hr style="height:1.75px;border:none;color:#4682B4;background-color:#4682B4;" />

In [4]:
#2.1 Number of rows and columns in the dataset
rows, columns= data.shape
print(f"The dataset contains {rows} rows and {columns} columns.\n")

The dataset contains 50024 rows and 13 columns.



In [5]:
#2.2 Column names
print(data.columns)

Index(['User_ID', 'Age', 'Gender', 'Occupation', 'Device_Type',
       'Daily_Phone_Hours', 'Social_Media_Hours', 'Work_Productivity_Score',
       'Sleep_Hours', 'Stress_Level', 'App_Usage_Count',
       'Caffeine_Intake_Cups', 'Weekend_Screen_Time_Hours'],
      dtype='object')


In [6]:
#2.3 Data types of each column 
print("\nData types:")
print (data.dtypes)


Data types:
User_ID                       object
Age                            int64
Gender                        object
Occupation                    object
Device_Type                   object
Daily_Phone_Hours            float64
Social_Media_Hours           float64
Work_Productivity_Score        int64
Sleep_Hours                  float64
Stress_Level                 float64
App_Usage_Count                int64
Caffeine_Intake_Cups           int64
Weekend_Screen_Time_Hours    float64
dtype: object


In [7]:
#2.4 Initial observations
data.head()

,User_ID,Age,Gender,Occupation,Device_Type,Daily_Phone_Hours,Social_Media_Hours,Work_Productivity_Score,Sleep_Hours,Stress_Level,App_Usage_Count,Caffeine_Intake_Cups,Weekend_Screen_Time_Hours
0,U1,58,Male,Professional,Android,1.3,6.7,6,8.8,NaN,42,1,8.7
1,U2,25,Male,Professional,Android,1.2,1.5,5,6.4,1.0,51,3,5.1
2,U3,19,Male,Student,iOS,5.3,5.7,5,9.0,4.0,14,5,6.3
3,U4,35,Female,Business Owner,iOS,5.8,2.5,2,5.7,3.0,36,6,12.8
4,U5,33,Male,Freelancer,Android,7.9,1.3,4,5.7,3.0,37,5,9.9


In [8]:
data.tail()

,User_ID,Age,Gender,Occupation,Device_Type,Daily_Phone_Hours,Social_Media_Hours,Work_Productivity_Score,Sleep_Hours,Stress_Level,App_Usage_Count,Caffeine_Intake_Cups,Weekend_Screen_Time_Hours
50019,U49863,40,Other,Student,Android,10.9,4.8,9,6.6,1.0,15,3,5.1
50020,U49864,20,Female,Professional,Android,2.3,1.0,6,8.7,2.0,42,3,7.3
50021,U49865,20,Male,Student,iOS,10.2,7.2,7,4.4,6.0,43,0,13.2
50022,U49866,20,Male,Business Owner,iOS,10.0,3.5,7,6.3,8.0,59,4,6.9
50023,U49867,37,Other,Freelancer,iOS,1.4,2.7,3,7.8,5.0,46,0,10.8


In [9]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50024 entries, 0 to 50023
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   User_ID                    50024 non-null  object 
 1   Age                        50024 non-null  int64  
 2   Gender                     50024 non-null  object 
 3   Occupation                 50024 non-null  object 
 4   Device_Type                50024 non-null  object 
 5   Daily_Phone_Hours          50024 non-null  float64
 6   Social_Media_Hours         50024 non-null  float64
 7   Work_Productivity_Score    50024 non-null  int64  
 8   Sleep_Hours                49997 non-null  float64
 9   Stress_Level               49998 non-null  float64
 10  App_Usage_Count            50024 non-null  int64  
 11  Caffeine_Intake_Cups       50024 non-null  int64  
 12  Weekend_Screen_Time_Hours  50024 non-null  float64
dtypes: float64(5), int64(4), object(4)
memory usag

In [10]:
print("\nSummery statistics:")
data.describe()


Summery statistics:


,Age,Daily_Phone_Hours,Social_Media_Hours,Work_Productivity_Score,Sleep_Hours,Stress_Level,App_Usage_Count,Caffeine_Intake_Cups,Weekend_Screen_Time_Hours
count,50024.000000,50024.000000,50024.000000,50024.000000,49997.000000,49998.000000,50024.000000,50024.000000,50024.000000
mean,39.034124,6.509340,4.267454,5.503718,6.497818,5.504740,32.439469,3.004698,8.007229
std,12.415810,3.171067,2.164829,2.874736,1.449596,2.871066,16.121193,1.996568,3.461106
min,18.000000,1.000000,0.500000,1.000000,4.000000,1.000000,5.000000,0.000000,2.000000
25%,28.000000,3.800000,2.400000,3.000000,5.200000,3.000000,18.000000,1.000000,5.000000
50%,39.000000,6.500000,4.300000,5.500000,6.500000,6.000000,32.000000,3.000000,8.000000
75%,50.000000,9.200000,6.100000,8.000000,7.800000,8.000000,46.000000,5.000000,11.000000
max,60.000000,12.000000,8.000000,10.000000,9.000000,10.000000,60.000000,6.000000,14.000000


### <font color='#4682B4'> Summery of Initial Overview (Pre-Cleaning) </font>
<hr style="height:1.75px;border:none;color:#4682B4;background-color:#4682B4;" />

- The dataset contains **50024 rows × 13 columns**, covering variables such as screen time, sleep hours, stress levels, productivity, caffeine intake, weekend screen time, and demographics.  
- **Data types** are consistent: numeric for continuous measures, categorical for usage/stress groups, gender, and occupation.  
- **Missing values check:** Some columns show, which will be handled in the data cleaning step.  
- **Head/Tail inspection:** Confirms the dataset is structured correctly and values fall within expected ranges.  
- **Describe():** Early stats highlight variation in screen time and sleep, with stress and productivity showing potential links.  
- **Info():** Confirms column types and non-null counts, ensuring readiness for cleaning and deeper analysis.  

These initial checks validate the dataset structure and highlight areas that need attention before proceeding to cleaning and detailed EDA.

### <font color='#4682B4'> 3.  Data Pre-processing </font>
<hr style="height:1.75px;border:none;color:#4682B4;background-color:#4682B4;" />

#### <font color='#4682B4'> 3.1.  Handling missing values </font>

In [12]:
#Checking missing values 

print(data.isnull().sum())

User_ID                       0
Age                           0
Gender                        0
Occupation                    0
Device_Type                   0
Daily_Phone_Hours             0
Social_Media_Hours            0
Work_Productivity_Score       0
Sleep_Hours                  27
Stress_Level                 26
App_Usage_Count               0
Caffeine_Intake_Cups          0
Weekend_Screen_Time_Hours     0
dtype: int64


In [13]:
#Handling missing values using median 

data['Sleep_Hours']= data['Sleep_Hours'].fillna(data['Sleep_Hours'].median())

data['Stress_Level']= data['Stress_Level'].fillna(data['Stress_Level'].median())

In [14]:
#Checking missing values are resolved

print(data.isnull().sum())

User_ID                      0
Age                          0
Gender                       0
Occupation                   0
Device_Type                  0
Daily_Phone_Hours            0
Social_Media_Hours           0
Work_Productivity_Score      0
Sleep_Hours                  0
Stress_Level                 0
App_Usage_Count              0
Caffeine_Intake_Cups         0
Weekend_Screen_Time_Hours    0
dtype: int64


#### <font color='#4682B4'> 3.2.  Removing duplicates </font>

In [15]:
#checking for duplicate rows

duplicates_count= data.duplicated().sum()
print("Number of duplicate rows:",duplicates_count)

Number of duplicate rows: 24


In [16]:
#Removing duplicate rows

data=data.drop_duplicates()

In [17]:
#Checking duplicate values resolved

duplicates_count= data.duplicated().sum()
print("Number of duplicate rows:",duplicates_count)

Number of duplicate rows: 0


#### <font color='#4682B4'> 3.3.  Correcting data types </font>

In [18]:
#Gender, Occupation, and Device Type are categorical fields.  
#Converting them from object to category type to improve efficiency and analysis.

categorical_cols = ['Gender', 'Occupation', 'Device_Type']
for col in categorical_cols:
    data[col] = data[col].astype('category')

In [19]:
print("\nData types:")
print (data.dtypes)


Data types:
User_ID                        object
Age                             int64
Gender                       category
Occupation                   category
Device_Type                  category
Daily_Phone_Hours             float64
Social_Media_Hours            float64
Work_Productivity_Score         int64
Sleep_Hours                   float64
Stress_Level                  float64
App_Usage_Count                 int64
Caffeine_Intake_Cups            int64
Weekend_Screen_Time_Hours     float64
dtype: object


#### <font color='#4682B4'> 3.4.  Creating derived columns  </font>

In [20]:
# A. Total daily screen time
data['Total_Screen_Time'] = data['Daily_Phone_Hours'] + data['Social_Media_Hours']

In [21]:
# B. Usage Categorization
def get_usage(hours):
    if hours <= 4: return 'Light'
    elif hours <= 8: return 'Moderate'
    else: return 'Heavy'
data['Usage_Category'] = data['Total_Screen_Time'].apply(get_usage)

In [22]:
# C. Stress Categorization
def get_stress(level):
    if level <= 3: return 'Low'
    elif level <= 7: return 'Moderate'
    else: return 'High'
data['Stress_Category'] = data['Stress_Level'].apply(get_stress)

In [23]:
# D. Sleep Efficiency (%)
data['Sleep_Efficiency'] = (data['Sleep_Hours'] / 24) * 100

In [24]:
# E. Digital Balance Score
data['Digital_Balance_Score'] = (data['Sleep_Hours'] + data['Work_Productivity_Score']) / (data['Total_Screen_Time'] + 1)

In [25]:
print(data.columns)

Index(['User_ID', 'Age', 'Gender', 'Occupation', 'Device_Type',
       'Daily_Phone_Hours', 'Social_Media_Hours', 'Work_Productivity_Score',
       'Sleep_Hours', 'Stress_Level', 'App_Usage_Count',
       'Caffeine_Intake_Cups', 'Weekend_Screen_Time_Hours',
       'Total_Screen_Time', 'Usage_Category', 'Stress_Category',
       'Sleep_Efficiency', 'Digital_Balance_Score'],
      dtype='object')


#### <font color='#4682B4'> 3.5.  Filtering or aggregating data  </font>

In [26]:
# 1. Filtering (Removing Outliers & Errors) 
# ensuring sleep is realistic (0-18 hours) and screen time is not negative.
data_cleaned = data[(data['Sleep_Hours'] >= 0) & (data['Sleep_Hours'] <= 18)].copy()
data_cleaned = data_cleaned[data_cleaned['Total_Screen_Time'] >= 0]

In [27]:
# 2. Final Aggregation (Data Quality Check) 
quality_check = data_cleaned.groupby('Usage_Category').agg({
    'User_ID': 'count',
    'Total_Screen_Time': ['min', 'max', 'mean'],
    'Digital_Balance_Score': 'mean'
}).round(2)

### <font color='#4682B4'> Final Data Preview </font>
<hr style="height:1.75px;border:none;color:#4682B4;background-color:#4682B4;" />

After completing all preprocessing steps (cleaning, type corrections, derived columns, filtering, aggregation), here’s the updated dataset:

In [30]:
data.head()

,User_ID,Age,Gender,Occupation,Device_Type,Daily_Phone_Hours,Social_Media_Hours,Work_Productivity_Score,Sleep_Hours,Stress_Level,App_Usage_Count,Caffeine_Intake_Cups,Weekend_Screen_Time_Hours,Total_Screen_Time,Usage_Category,Stress_Category,Sleep_Efficiency,Digital_Balance_Score
0,U1,58,Male,Professional,Android,1.3,6.7,6,8.8,6.0,42,1,8.7,8.0,Moderate,Moderate,36.666667,1.644444
1,U2,25,Male,Professional,Android,1.2,1.5,5,6.4,1.0,51,3,5.1,2.7,Light,Low,26.666667,3.081081
2,U3,19,Male,Student,iOS,5.3,5.7,5,9.0,4.0,14,5,6.3,11.0,Heavy,Moderate,37.500000,1.166667
3,U4,35,Female,Business Owner,iOS,5.8,2.5,2,5.7,3.0,36,6,12.8,8.3,Heavy,Low,23.750000,0.827957
4,U5,33,Male,Freelancer,Android,7.9,1.3,4,5.7,3.0,37,5,9.9,9.2,Heavy,Low,23.750000,0.950980
